### ISSUE TITLE
Query on Table 4

### ISSUE THREAD

**Issue by robinsongh381** [facebookresearch/DPR#126]
Hello !

I could not find any explicit mention from the paper that whether the DPR result for the Table 4 is from **dev** or **test** set.

I suspect, reader models were examined on test datasets. 
- If so, do they correspond to a json output which is generated upon running `dense_retriever.py` against `data/data/retriever/qas/{nq, trivia, squad1}-test.csv` ?

Thank you

**Comment by vlad-karpukhin** [comment]
Hi, 
Results in the paper are from test sets (official dev for NQ). Our dev split for NQ is our own and doesn't correspond to that mentioned in google papers like ORQA or REALM.

**Comment by robinsongh381** [comment]
Alright, thanks.

Could you please share preprocessed test sets ? Or a code which preprocesses official test sets into your desired format.
This would be very much helpful 

Thanks

**Comment by vlad-karpukhin** [comment]
Hi @robinsongh381 ,
Sorry for a late reply, just noted your follow-up questions.

1. All QAS (questions & answers) data files are only for inference/evaluation, they don't have any passages, just questions & answers. So one can't use them for any training. All our train sets used for training are subsets (sometimes full sets) of the official train sets. All our 'qas' train sets are official train splits but without passages.
For some datasets like NQ, we lost part of the official train samples because of gold passages mapping.

2. I don't understand this question. Can you elaborate?

3. As of NQ open - it wasn't officially ready for downloading when we were doing DPR. We just followed exact same process described in ORQA paper to extract this dataset from full NQ data and got same total amount of (train+dev) & test splits like in the google's paper, just our dev split is different (i.e. it has same amount of samples but they are different in general). 
dense_retriever.py doesn't know anything about official ctx, it just returns best candidates from our wikipedia index, which has nothing to do with 'official' NQ ctxs (or as they call them candidate spans).

4. The usage of gold_passages_src is NQ-only heuristic and is used just to en-force injecting gold ctx+ into the reader training process and select other ctx+ and ctx- based on the link to the gold wikipedia page. "your provided gold_passages_src would no longer valid for "official" dev set" - yes but this task's NQ dev set is only used for checkpoint selection and we don't use its results for any comparison with other papers.

Please note that gold_passages_src used for reader data pre-processing is not the same step when I mention "gold ctx+ mapping to our wikipedia split"

To summarize NQ splits processing pipeline:

1. Full official NQ train set -> filter open domain task samples -> split to train & dev splits -> these are 'qas' files.
2.  Do gold ctx+ mapping for train/dev sets from #1 to generate bi-encoder training data. this step looses ~25% of samples. These are train&dev DPR json files.
3.  Full official NQ dev set -> filter open domain task samples ->  this is test split 'qas' file.
